### PR TITLE
TG: Adding datadog metrics and monitors

### PR DESCRIPTION
**Pull request by janorivera**
## Description

This code adds a metric and a monitor per channel when enabled.
This was an easy setup so I just went ahead and implement it to test it out :) 
Reviewers for visibility.
### Checklist
- [ ] Corresponding issue has been opened
- [ ] New tests added
- [ ] Feature flags added
- [ ] Strings are localized
- [ ] Tested for chat contacts
- [ ] Tested for call contacts

### Other Related Issues

None

### AFTER YOU MERGE

1. Cut a release tag using the Github workflow. Wait for it to complete and notify in the #aselo-deploys Slack channel.
2. Comment on the ticket with the release tag version AND any additional instructions required to configure an environment to test the changes.
3. Only then move the ticket into the QA column in JIRA

You are responsible for ensuring the above steps are completed. If you move a ticket into QA without advising what version to test, the QA team will assume the latest tag has the changes. If it does not, the following confusion is on you! :-P

### CODE REVIEW

**Comment by janorivera** [comment]
@stephenhand I know we spoke about this, but we didn't agree to actually do it or when. I was already looking into how to manually creating these monitors and I saw that datadog offer the script on terraform to implement it so I just went ahead and implemented the datadog module to create these on our twilio-iac folders. 
It wasn't difficult to set this up! and it is just something that it might be worth trying it out and see if it brings value.
It seems that each metric has a cost of 0.05 USD so implementing 20 extra metrics would be just 1 USD of additional Datadog costs.
Anyways, sorry for jumping ahead :)
Let me know if you want to review it or if you have any comments. 
{DA7912CB-75FD-4926-9275-A9A0368C68C7}

File: twilio-iac/terraform-modules/datadog/v1/main.tf [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  required_providers {
+    datadog = {
+      source = "datadog/datadog"
+      version = "3.46.0"
+    }
+  }
+}
+
+
+resource "datadog_logs_metric" "channel_logs_metric" {
```
Comment (stephenhand) [inline twilio-iac/terraform-modules/datadog/v1/main.tf:11]: Can we add a "terraform' tag to each of these resources, so datadog users know they are set up by terraform and changes they make in data dog will be reverted?

**Review by stephenhand — APPROVED** [review]
One minor comment, otherwise lgtm

File: twilio-iac/terraform-modules/datadog/v1/main.tf (hunk 1 above)
Comment (janorivera) [reply twilio-iac/terraform-modules/datadog/v1/main.tf:11]: sure no problem!

**Pull request merged**